DEVELOPER.md

```diff
@@ -0,0 +1,374 @@
+# Developer Guide for TestSys
+
+## What is TestSys
+
+TestSys is a framework that leverages Kubernetes (k8s) to create resources and run tests.
+To learn more about how it works, see the [design](design/DESIGN.md) document.
+
+### Navigating the `bottlerocket-test-system` repo
+
+#### [Makefile.toml](Makefile.toml)
+
+TestSys uses make to unify the logic needed to build images.
+Any image can be built using `make <desired image>`.
+To make all bottlerocket related images, `make images` can be used, then `make tag-images` can be used to give all images the same tag and `make publish-images` can be used to publish all images.
+
+#### [Dockerfile](Dockerfile)
+
+Testsys relies on docker images to run tests and create resources.
+The Dockerfile contains the logic needed to create all bottlerocket related images.
+
+#### [Agent](agent)
+
+Testsys is broken up into 3 main things: the controller, test agent, and resource agent.
+
+[agent-common](agent/agent-common) contains logic that is related to both the test and resource agents.
+
+[test-agent](agent/test-agent) provides the framework for creating tests.
+A test is a program that runs a test and reports its outcome by updating the TestSys Test CRD object's status fields. 
+The test agent is packaged into a container image and the controller runs it in a pod as a Kubernetes Job. 
+To create a test agent, the [`Runner` trait](agent/test-agent/src/lib.rs#L51) needs to be implemented.
+Check out the [example test agent](agent/test-agent/examples/example_test_agent/main.rs) to see how to create a test agent.
+
+[resource-agent](agent/resource-agent) provides the framework for creating resources.
+A ResourceAgent can create and destroy external resources (such as Kubernetes clusters or compute instances) that are needed for a test. 
```

Comment:
nit: markup for code-related terms and elsewhere as well
```suggestion
A `ResourceAgent` can create and destroy external resources (such as Kubernetes clusters or compute instances) that are needed for a test. 
```

We also refer to the test agent plainly as "test agent" right above, so this can just be 
```suggestion
A resource agent can create and destroy external resources (such as Kubernetes clusters or compute instances) that are needed for a test. 
```